### ISSUE TITLE
java UnsatisfiedLinkError when new CassandraValueMergeOperator()

### ISSUE THREAD

**Issue by javeme** [facebook/rocksdb#3206]
java UnsatisfiedLinkError when new CassandraValueMergeOperator()

```
java.lang.UnsatisfiedLinkError: org.rocksdb.CassandraValueMergeOperator.newSharedCassandraValueMergeOperator()J
	at org.rocksdb.CassandraValueMergeOperator.newSharedCassandraValueMergeOperator(Native Method)
	at org.rocksdb.CassandraValueMergeOperator.<init>(CassandraValueMergeOperator.java:14)
	at com.baidu.hugegraph.backend.store.rocksdb.RocksDBSessions.initOptions(RocksDBSessions.java:151)
```

Happens in version 5.8.6 on Mac.

**Comment by yiwu-arbug** [comment]
@sagar0 Any idea?

**Comment by adamretter** [comment]
@yiwu-arbug @javeme It is missing from `JNI_NATIVE_SOURCES` in both `src.mk` and `java/CMakeLists.txt`.

Seems that https://github.com/facebook/rocksdb/pull/2289 was incomplete.

**Comment by sagar0** [comment]
This issue is fixed in #2791 , and will go into 5.9 release. 
@javeme May I ask why you are using CassandraValueMergeOperator? It is not useful outside Cassandra on RocksDB (Rocksandra) project : https://github.com/instagram/cassandra/tree/rocks_3.0
